ServerSideScript.py

```diff
@@ -0,0 +1,170 @@
+#handle both getting information from POST and retrieval of info from _grab
+
+#needs to be run in one instance of cmd, then run TestScript in another window to conduct tests.
+    #figure out how the ServerSideScript should work, in terms of classes/functions structures
+
+import datetime
+from flask import Flask, request
+import json
+from http.server import BaseHTTPRequestHandler, HTTPServer
+import time
+import pickle
+
+ServerSideScript = Flask(__name__)
+
+#this should be the place where the job info is stored:
+    #'<job_id>': {'fn': <fn>, 'func_args': <*args>, 'func_kwargs': <**kwargs>, 
+    #   'job_id': <job_id>, 'memory': <memory>, 'priority': <priority>, 
+    #   'retries': <retries>, 'run_now': run_now}
+#change data storage to pickle format, have the function name pickled
+    #convert each job back into object, and figure out which package and modules it is in, and load it for me
+
+#pickled and non pickled:
+    #only 'fn', 'func_args', and 'func_kwargs' will be pickled
+data_for_jobs = {'19838182.391829': {'fn': 'placeHolder', 'func_args': [], 
+    'func_kwargs': {}, 'job_id': 19838182.391829, 'memory': 1024, 
+    'priority': 2, 'retries': 1, 'run_now': True, 'complete': False, 
+    'running': False, 'cancelled': False, 'exception': None}}
+
+@ServerSideScript.route('/')
+def default():
+    return 'Up and running!'
+
+#sends back unique id for a function in str form so its static
+def function_encoder(fn):
+    return str(datetime.datetime.now().timestamp())
+
+server_running = True
+#make sure I have a function to manage all the running of the jobs
+                #######currently not suitable
+def job_manager():
+    while (server_running):
+        #every iteration of this large loop is one job ran
+        #adjust sleep time arbitrarily, or by system constraints
+        time.sleep(2.4)
+        if (data_for_jobs is not None):
+            #update the viable jobs list
+            runnable_jobs = dict()        
+            for job in data_for_jobs.items():

```

Comment:
Write it like
`for job_id, job in data_for_jobs.items()`

Reply:
the access to data_for_jobs needs to be guarded by locks, to pervent simultaneous write access (which modify data_for_jobs)